sklearn/decomposition/_dict_learning.py

```diff
@@ -802,7 +788,7 @@ def dict_learning_online(X, n_components=2, *, alpha=1, n_iter=100,
     else:
         X_train = X
 
-    dictionary = check_array(dictionary.T, order='F', dtype=np.float64,
+    dictionary = check_array(dictionary, order='F', dtype=np.float64,
```

Comment:
We want the dictionary to be fortran, not its transpose. Makes the sparse coding faster